Macros.h

```diff
@@ -61,6 +61,18 @@
 
 #endif /* HAVE_ATTR_ALLOC_SIZE */
 
+#ifdef HAVE_ATTR_ACCESS
+
+#define ATTR_ACCESS2(mode, ref)         __attribute__((access (mode, ref)))
+#define ATTR_ACCESS3(mode, ref, size)   __attribute__((access (mode, ref, size)))
```

Comment:
Why not pre-define `ATTR_ACCESS_R`, `ATTR_ACCESS_RW`, and `ATTR_ACCESS_W`?

Reply:
I can see why you define two versions of the macro `ATTR_ACCESS2`, `ATTR_ACCESS3`, but consider using `__VA_ARGS__`.

```suggestion
// Usage: ATTR_ACCESS(mode, refIndex, sizeIndex) while sizeIndex is optional
#define ATTR_ACCESS(mode, ...)          __attribute__((access (mode, __VA_ARGS__)))
```

Or, as @BenBE suggested:

```suggestion
// Usage: ATTR_ACCESS_X(refIndex, sizeIndex) while sizeIndex is optional
#define ATTR_ACCESS_R(...)              __attribute__((access (read_only, __VA_ARGS__)))
#define ATTR_ACCESS_RW(...)             __attribute__((access (read_write, __VA_ARGS__)))
#define ATTR_ACCESS_W(...)              __attribute__((access (write_only, __VA_ARGS__)))
```

Reply:
I'd prefer to have 6 macros instead, as the `__VA_ARGS__` semantic is not intended and misleading in this case. Yes, that's some more macros, but clarity is better in this case. I even could see a reason to keep all 8 variants:
```suggestion
#define ATTR_ACCESS2(mode, ref)         __attribute__((access (mode, ref)))
#define ATTR_ACCESS3(mode, ref, size)   __attribute__((access (mode, ref, size)))

#define ATTR_ACCESS2_R(ref)         ATTR_ACCESS2(read_only, ref)
#define ATTR_ACCESS3_R(ref, size)   ATTR_ACCESS3(read_only, ref, size)

#define ATTR_ACCESS2_RW(ref)         ATTR_ACCESS2(read_write, ref)
#define ATTR_ACCESS3_RW(ref, size)   ATTR_ACCESS3(read_write, ref, size)

#define ATTR_ACCESS2_W(ref)         ATTR_ACCESS2(write_only, ref)
#define ATTR_ACCESS3_W(ref, size)   ATTR_ACCESS3(write_only, ref, size)
```